src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp

```diff
@@ -305,150 +305,134 @@ ov::Tensor create_and_fill_tensor_consistently(const ov::element::Type element_t
 
 constexpr double eps = std::numeric_limits<double>::epsilon();
 
-inline double less(double a, double b) {
+template <typename aT, typename bT>
+inline double less(aT a, bT b) {
     return std::fabs(a - b) > eps && a < b;
 }
 
 inline double less_or_equal(double a, double b) {
     bool res = true;
     if (std::isnan(a) || std::isnan(b)) {
         res = false;
+    } else if (std::isinf(b) && std::isinf(b)) {
+        res = true;
     } else if (std::isinf(b) && b > 0) {
-        // b is grater than any number or eq the +Inf
+        // b is greater than any number or eq the +Inf
         res = true;
     } else if (std::isinf(a) && a > 0) {
         res = false;
     } else {
-        res = (std::fabs(b - a) <= (std::fmax(std::fabs(a), std::fabs(b)) * eps) || a < b);
+        res = std::fabs(b - a) <= eps || a <= b;
     }
-    double eq_midle_res = std::fabs(b - a);
-    bool eq_res = (std::fabs(b - a) <= (std::fmax(std::fabs(a), std::fabs(b)) * eps));
     return res;
 }
 
-struct Error {
-    double max = 0.;
-    double mean = 0.;
-    size_t max_coordinate = 0;
-    size_t count = 0;
-    double threshold;
+class Error {
+protected:
+    struct IncorrectValue {
+        size_t coordinate;
+        double actual_value, expected_value, threshold;
+
+        IncorrectValue(double in_actual_value, double in_expected_value, double in_threshold, size_t in_coordinate)
```

Comment:
to store both: rel and abs threshold

Reply:
here they are not changing and are saved by the constructor, only the calculated_abs_threshold is different